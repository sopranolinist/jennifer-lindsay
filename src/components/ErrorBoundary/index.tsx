import {
  createStyles,
  Theme,
  withStyles,
  WithStyles
} from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import React, { Component, ReactChild } from 'react';
import DocumentTitle from 'react-document-title';
import { getPageTitle } from '../../helpers/DataHelper';

const styles = (theme: Theme) =>
  createStyles({
    root: {
      alignItems: 'baseline',
      padding: theme.spacing(3),
      paddingTop: theme.spacing(12),
      flexGrow: 1,
      flexDirection: 'column',
      display: 'flex',
      textAlign: 'left'
    },
    errorHeader: {
      color: theme.palette.error['main'],
      marginBottom: theme.spacing(2)
    },
    errorMessage: {
      marginBottom: theme.spacing(2)
    }
  });

interface Props extends WithStyles<typeof styles> {
  children?: ReactChild;
  hasError?: string;
}
interface State {
  error: boolean;
}

class ErrorBoundary extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { error: false };
  }

  static getDerivedStateFromError(error: Error | null) {
    return { error: true };
  }

  componentDidCatch(error: Error | null, info: object) {
    // console.log('ErrorBoundary.componentDidCatch', error, info);
  }

  render() {
    const pageTitle = 'Error';
    const { error } = this.state;
    const { children, hasError, classes } = this.props;
    if (!error && !hasError) return children;
    return (
      <div className={classes.root}>
        <DocumentTitle title={getPageTitle(pageTitle)} />
        <Typography component="h1" variant="h6" className={classes.errorHeader}>
          Error
        </Typography>
        <Typography
          component="span"
          variant="body1"
          className={classes.errorMessage}
        >
          The site has encountered an error{hasError && ':'}
        </Typography>
        {hasError && (
          <Typography
            component="span"
            variant="body1"
            className={classes.errorMessage}
          >
            {hasError}
          </Typography>
        )}
      </div>
    );
  }
}

export default withStyles(styles)(ErrorBoundary);
