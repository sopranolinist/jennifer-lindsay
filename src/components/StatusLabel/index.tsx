import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import classnames from 'classnames';
import React from 'react';
import { palette } from '../../theme/materialUi';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    smStatusBox: {
      width: '90px',
      height: '30px',
      padding: '5px 5px',
      borderRadius: '8px',
      display: 'block',
      textAlign: 'center',
      fontWeight: 600
    },
    mdStatusBox: {
      width: '100px',
      height: '30px',
      padding: '5px 10px',
      borderRadius: '8px',
      display: 'block',
      textAlign: 'center',
      fontWeight: 'bold'
    },
    lgStatusBox: {
      width: '140px',
      height: '40px',
      padding: '5px 10px',
      borderRadius: '8px',
      display: 'block',
      textAlign: 'center',
      fontWeight: 'bold'
    },
    statusApproved: {
      backgroundColor: palette.status.approved
    },
    statusDenied: {
      backgroundColor: palette.status.denied
    },
    statusPending: {
      backgroundColor: palette.status.pending
    },
    statusCancelled: {
      backgroundColor: palette.status.cancelled
    }
  })
);

export enum BoxSize {
  sm = 'sm',
  md = 'md',
  lg = 'lg'
}

interface StatusLabelProps {
  status: string;
  boxSize: BoxSize;
}

const StatusLabel = (props: StatusLabelProps) => {
  const { status, boxSize } = props;
  const classes = useStyles();

  return (
    <span
      className={classnames(
        boxSize === 'sm'
          ? classes.smStatusBox
          : boxSize === 'md'
            ? classes.mdStatusBox
            : classes.lgStatusBox,
        status === 'APPROVED'
          ? classes.statusApproved
          : status === 'DENIED'
            ? classes.statusDenied
            : status === 'CANCELLED'
              ? classes.statusCancelled
              : classes.statusPending
      )}
    >
      {status}
    </span>
  );
};

export default StatusLabel;
