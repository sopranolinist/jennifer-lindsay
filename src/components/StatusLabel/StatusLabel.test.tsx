import { render } from '@testing-library/react';
import { advanceTo, clear } from 'jest-date-mock';
import React from 'react';
import {
  authExamples,
  getFullyWrappedComponent,
  getJssWrappedComponent,
  getRouterWrappedComponent,
  mockConsole
} from '../../helpers/TestHelper';
import ErrorBoundary from '../ErrorBoundary';
import StatusLabel, { BoxSize } from './';

interface Props {}

describe('Component - Footer', () => {
  beforeEach(() => {
    mockConsole();
    advanceTo(1558461651876);
  });
  afterEach(() => {
    clear();
  });

  test('should enforce auth context', async done => {
    const { container } = render(
      getJssWrappedComponent(
        getRouterWrappedComponent(
          <ErrorBoundary>
            <StatusLabel status="DENIED" boxSize={BoxSize.sm} />
          </ErrorBoundary>
        )
      )
    );
    expect(container).toMatchSnapshot();
    done();
  });

  test('renders the StatusLabel components', async done => {
    const { authContextProps, authObjProps } = authExamples.signedInAuthorized;
    const { container } = render(
      getFullyWrappedComponent(
        <StatusLabel status="APPROVED" boxSize={BoxSize.md} />,
        { authContextProps, authObjProps }
      )
    );
    expect(container).toMatchSnapshot();
    done();
  });
});
