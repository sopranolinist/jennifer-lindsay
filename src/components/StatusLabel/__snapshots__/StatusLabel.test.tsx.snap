// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Footer renders the StatusLabel components 1`] = `
<div>
  <span
    class="makeStyles-mdStatusBox-5 makeStyles-statusApproved-7"
  >
    APPROVED
  </span>
</div>
`;

exports[`Component - Footer should enforce auth context 1`] = `
<div>
  <span
    class="makeStyles-smStatusBox-4 makeStyles-statusDenied-8"
  >
    DENIED
  </span>
</div>
`;
